src/main/scala/mesosphere/marathon/core/appinfo/GroupInfoService.scala

```diff
@@ -0,0 +1,17 @@
+package mesosphere.marathon.core.appinfo
+
+import mesosphere.marathon.state.Group
+
+import scala.concurrent.Future
+
+/**
+  * Queries for extended group information.
+  */
+trait GroupInfoService {
+
+  /**
+    * Query info for an existing group.
+    */
+  def queryForGroup(group: Group, appEmbed: Set[AppInfo.Embed], groupEmbed: Set[GroupInfo.Embed]): Future[GroupInfo]
```

Comment:
For me, it feels a bit unnatural that you pass the group to this method. Is it because we do the security stuff in the resource layer? Wouldn't it be better to pass the "Identity" into this method and do the filtering here?